lib/network.py

```diff
@@ -1486,7 +1491,6 @@ def run(self):
         while self.is_running():
             self.maintain_sockets()
             self.wait_on_sockets()
-            self.maintain_requests()
```

Comment:
How do we detect a server that has gone out to lunch now and that isn't responding?   You know, like the server equivalent of `/dev/null`, basically?  A black hole where you send requests and never get a reply.... How is that handled now?

Reply:
Yeah I didn't explain fully. In `maintain_sockets` there is a `has_timed_out` check which waits for 10s after last request queued, if there are still unanswered requests *and no data has been received* -- that last conditional being the missing thing in maintain_requests.

Reply:
Hmm.  I see. Ok so that handles the "general black hole" case well enough.

Actually this code never realizes if a particular request has taken too long.. but that is not new with this change.  Maybe we need to handle that later but it would add complexity... (if we migrate to asyncio it would be trivial to fix since we can enqueue then "on the event loop" in some way and have the timeout come naturally on a per-request basis).

Ok.. I'm satisfied.. I think.

Curious that this code found a need before to `maintain_requests` -- when it was also checking `has_timed_out` anyway.  Hmm.  Why do you think it had both before?  Just general sloppiness?

Reply:
Yeah I was looking through the code history, and `maintain_requests` used to be a much bigger function from ThomasV back a while ago. I guess it got slowly whittled down to this, and perhaps the other timeout was added independently.

One avenue I was thinking might be worth pursuing is a timeout of this kind but based on `last_send` rather than on `request_time`, since `last_send` gets updated more often.

Reply:
Hmm. No idea. Thinking about this makes my brain melt.  Your call here. :)

Reply:
Eh let's just leave it like this for now. Anyway in future if we need to bring in something like this it belongs in `has_timed_out()` anyway, and we don't need maintain_requests.